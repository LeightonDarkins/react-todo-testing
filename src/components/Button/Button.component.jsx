import React from 'react'
import PropTypes from 'prop-types'

import './Button.scss'

const AddButton = ({ label, type, small, disabled, buttonClicked }) => {
  const sizeClass = small ? 'Button--small' : ''

  const buttonClass = () => {
    switch (type) {
      case 'add':
        return `Button--${type}`
      case 'delete':
        return `Button--${type}`
      default:
        return 'Button'
    }
  }

  const cssClass = `${buttonClass()} ${sizeClass}`

  if (disabled) {
    return <button className={cssClass} disabled>{label}</button>
  } else {
    return <button className={cssClass} onClick={buttonClicked}>{label}</button>
  }
}

AddButton.propTypes = {
  label: PropTypes.string,
  type: PropTypes.string,
  small: PropTypes.bool,
  disabled: PropTypes.bool,
  buttonClicked: PropTypes.func.isRequired
}

module.exports = AddButton
