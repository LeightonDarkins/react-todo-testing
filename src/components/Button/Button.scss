.Button {
  font-size: 20px;
  text-transform: uppercase;
  height: 57px;
  width: 350px;
  background-color: #1A3B7A;
  color: white;
  font-weight: bold;
  border-radius: 5px;
  display: block;
  margin: 2.5px 0 2.5px 0;
}

.Button:hover {
  background-color: #081958;
  cursor: pointer;
}

.Button:focus {
  outline: 0;
  border: 2px solid #081958;
}

.Button:disabled {
  pointer-events: none;
  background-color: grey;
}

.Button--add {
  @extend .Button;

  background-color: #74BA00;
}

.Button--add:hover {
  background-color: #529800;
}

.Button--add:focus {
  border: 2px solid #529800;
}

.Button--delete {
  @extend .Button;

  background-color: #FF5916;
}

.Button--delete:hover {
  background-color: #DD3704;
}

.Button--delete:focus {
  border: 2px solid #DD3704;
}

.Button--small {
  @extend .Button;

  width: 175px;
}
