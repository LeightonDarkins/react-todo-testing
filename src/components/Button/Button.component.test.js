/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Button from './Button.component.jsx'

describe('Button', () => {
  it('propagates an event when clicked', () => {
    let mockButtonClicked = jest.fn()
    let wrapper = shallow(<Button buttonClicked={mockButtonClicked} />)

    wrapper.simulate('click')

    expect(mockButtonClicked).toHaveBeenCalledTimes(1)
  })

  it('a disabled button does not propagate an event when clicked', () => {
    let mockButtonClicked = jest.fn()
    let wrapper = shallow(<Button buttonClicked={mockButtonClicked} disabled />)

    wrapper.simulate('click')

    expect(mockButtonClicked).toHaveBeenCalledTimes(0)
  })

  it('renders a delete button', () => {
    const TYPE = 'delete'

    const tree = renderer
      .create(<Button buttonClicked={jest.fn()} type={TYPE} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders an add button', () => {
    const TYPE = 'add'

    const tree = renderer
      .create(<Button buttonClicked={jest.fn()} type={TYPE} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders a default button', () => {
    const tree = renderer
      .create(<Button buttonClicked={jest.fn()} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders a default, disabled button', () => {
    const tree = renderer
      .create(<Button buttonClicked={jest.fn()} disabled />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders a small delete button', () => {
    const tree = renderer
      .create(<Button buttonClicked={jest.fn()} type='delete' small />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
