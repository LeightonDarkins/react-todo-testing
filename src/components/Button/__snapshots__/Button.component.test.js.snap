// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a default button 1`] = `
<button
  className="Button "
  onClick={[MockFunction]}
/>
`;

exports[`Button renders a default, disabled button 1`] = `
<button
  className="Button "
  disabled={true}
/>
`;

exports[`Button renders a delete button 1`] = `
<button
  className="Button--delete "
  onClick={[MockFunction]}
/>
`;

exports[`Button renders a small delete button 1`] = `
<button
  className="Button--delete Button--small"
  onClick={[MockFunction]}
/>
`;

exports[`Button renders an add button 1`] = `
<button
  className="Button--add "
  onClick={[MockFunction]}
/>
`;
